axelrod/strategies/geller.py

```diff
@@ -47,27 +46,25 @@ class Geller(Player):
         'manipulates_state': False
     }
 
+    @staticmethod
+    def foil_strategy_inspection() -> Action:
+        return random_choice(0.5)
+
     def strategy(self, opponent: Player) -> Action:
         """
         Look at what the opponent will play in the next round and choose a strategy
         that gives the least jail time, which is is equivalent to playing the same
         strategy as that which the opponent will play.
         """
-        curframe = inspect.currentframe()
-        calframe = inspect.getouterframes(curframe, 2)
-        calname = calframe[1][3]
-        if calname == 'strategy':
-            return self.default()
-        else:
-            return opponent.strategy(self)
+
+        return inspect_strategy(self, opponent)
 
 
 class GellerCooperator(Geller):
     """Observes what the payer will do (like :code:`Geller`) but if unable to
     will cooperate.
     """
     name = 'Geller Cooperator'
-    default = lambda self: C
```

Comment:
the IDE balked at making a variable a lambda and recommended changing to a function.  so i did.